crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3190,39 +3173,50 @@ impl<'db> TypeInferenceBuilder<'db> {
         value_ty: Type<'db>,
         slice_ty: Type<'db>,
     ) -> Type<'db> {
+        fn iterator_at_index<T>(
```

Comment:
For slice expressions, I intend to move both `iterator_at_index` and `slice_at_index` to a separate module, which will then also include functionality to do something similar for slices.

Reply:
I decided to add that to this branch after all.

Reply:
And merged the two functions into a single one.